src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/AbstractCodeActionOrUserDiagnosticTest.cs

```diff
@@ -199,138 +330,6 @@ protected Task TestSmartTagGlyphTagsAsync(string initialMarkup, ImmutableArray<s
-            TestWorkspace workspace,
-            string expected,
-            ImmutableArray<CodeActionOperation> operations,
-            bool hasProjectChange,
-            ProjectId modifiedProjectId,
-            ImmutableArray<string> expectedFolders,
-            string expectedDocumentName)
-        {
-            var appliedChanges = ApplyOperationsAndGetSolution(workspace, operations);
-            var oldSolution = appliedChanges.Item1;
-            var newSolution = appliedChanges.Item2;
-
-            Document addedDocument = null;
-            if (!hasProjectChange)
-            {
-                addedDocument = SolutionUtilities.GetSingleAddedDocument(oldSolution, newSolution);
-            }
-            else
-            {
-                Assert.NotNull(modifiedProjectId);
-                addedDocument = newSolution.GetProject(modifiedProjectId).Documents.SingleOrDefault(doc => doc.Name == expectedDocumentName);
-            }
-
-            Assert.NotNull(addedDocument);
-
-            AssertEx.Equal(expectedFolders, addedDocument.Folders);
-            Assert.Equal(expectedDocumentName, addedDocument.Name);
-            Assert.Equal(expected, (await addedDocument.GetTextAsync()).ToString());
-
-            var editHandler = workspace.ExportProvider.GetExportedValue<ICodeActionEditHandlerService>();
-            if (!hasProjectChange)
-            {
-                // If there is just one document change then we expect the preview to be a WpfTextView
-                var content = (await editHandler.GetPreviews(workspace, operations, CancellationToken.None).GetPreviewsAsync())[0];
-                using (var diffView = content as DifferenceViewerPreview)
-                {
-                    Assert.NotNull(diffView.Viewer);
-                }
-            }
-            else
-            {
-                // If there are more changes than just the document we need to browse all the changes and get the document change
-                var contents = editHandler.GetPreviews(workspace, operations, CancellationToken.None);
-                var hasPreview = false;
-                var previews = await contents.GetPreviewsAsync();
-                if (previews != null)
-                {
-                    foreach (var preview in previews)
-                    {
-                        if (preview != null)
-                        {
-                            var diffView = preview as DifferenceViewerPreview;
-                            if (diffView?.Viewer != null)
-                            {
-                                hasPreview = true;
-                                diffView.Dispose();
-                                break;
-                            }
-                        }
-                    }
-                }
-
-                Assert.True(hasPreview);
-            }
-
-            return Tuple.Create(oldSolution, newSolution);
-        }
```

Comment:
why did all this cahnge?

Reply:
This file has been split into partial types and the part specific to core diagnostic/fixer tests is now used by both IDE test layer and CodeStyle test layer.